### PR TITLE
05 votes

### PR DESCRIPTION
**Pull request by Vondry66**

### CODE REVIEW

File: src/components/SingleReview.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const[isClicked,setIsClicked]=useState(false)
+    useEffect(()=>{
+        getSingleReview(review_id).then((rev)=>{
+            setReview(rev)
+            setVotes(rev.votes)
+        })
+    },[review_id])
+
+const handleVotes=()=>{
+    if(isClicked){
```
Comment (yalicee) [inline src/components/SingleReview.jsx]: You haven't interacted with the API!! If you refresh it stays on the same amounts of votes it was on before you voted. You need to use the patch endpoint we have

**Review by yalicee — CHANGES_REQUESTED** [review]
See comment :)

File: src/components/SingleReview.jsx (hunk 1 above)
Comment (Vondry66) [reply src/components/SingleReview.jsx]: Oh of course. Will do.

**Review by leekli — CHANGES_REQUESTED** [review]
Hey Filip,

So basically - when I click an article, nothing happens. It doesn't load the article, it gives me a blank page, this happens on different browsers.

Looking at your singlereview.jsx file, you've put your patchVotes login into the useEffect, this means it's trying increase votes when the page loads, this isn't right - you need that logic in its own function, the useEffect is basically failing and never loading an article for me. Your useEffect should only have the getSingleReview stuff in there, the patchVotes logic should be in a separate function not linked to useEffect.

Please go back to this ticket, it still needs quite a bit of work. Submit a helpdesk tomorrow if you're stuck.

**Review by leekli — APPROVED** [review]
Well done Filip, thanks for sorting this I can see it now works.

I'm happy for you to merge this branch.

I did have a problem though... when I tried to load the app on to the home page/home route, nothing loaded.... it looks like you're trying to import a component in App.js called 'Comments' which physically does not exist, I had to manually remove that line along with the comments route in App.js so that I could get your app to load.... normally I would have sent you the pull request back to fix, but I'd like you to move on. Please make sure you've fixed this issue before submitting the pull request for your next ticket

**Pull request merged**